### PR TITLE
feat: Add a link in the UI for WorkflowTemplate. Fixes #4760

### LINKED ISSUE #4760: If a Workflow, CronWorkflow, etc. uses a WorkflowTemplate as `workflowTemplateRef`, provide a link in the UI directly to that template
# Summary

What change needs making?

# Use Cases

When would you use this?

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by shubham4443**
Signed-off-by: Shubham Nazare <shubham4443@gmail.com>

Fixes #4760 

If a Workflow or CronWorkflow uses a WorkflowTemplate or ClusterWorkflowTemplate as workflowTemplateRef, a link is provided to redirect to that template.

For WorkflowTemplate kind -

image

For ClusterWorkflowTemplate kind -

image

### CODE REVIEW

**Comment by alexec** [comment]
Run “yarn lint”

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 1]
```diff
@@ -87,6 +87,99 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
...
+                }
+            },
+            {
+                title: 'Share',
+                iconClassName: 'fa fa-share-alt',
+                action: () => setSidePanel('share')
+            }
+        ];
+
+        if (cronWorkflow?.spec?.workflowSpec?.workflowTemplateRef) {
```
Comment (alexec) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:162]: I did not know about “?” until recently.

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx [hunk 2]
```diff
@@ -87,6 +87,99 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
...
+            const templateName = cronWorkflow.spec.workflowSpec.workflowTemplateRef.name;
+            const clusterScope = cronWorkflow.spec.workflowSpec.workflowTemplateRef.clusterScope;
+            const url: string = clusterScope ? `/cluster-workflow-templates/${templateName}` : `/workflow-templates/${cronWorkflow.metadata.namespace}/${templateName}`;
+
+            const templateLink: Link = {
+                name: 'Workflow Template Link',
```
Comment (alexec) [inline ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:168]: Please change to “Open Workflow Template” and use the same icon as left-hand-nav does.

File: ui/src/app/workflows/components/workflow-details/workflow-details.tsx [hunk 3]
```diff
@@ -129,6 +129,25 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
...
                 });
         }
+
+        if (workflow?.spec?.workflowTemplateRef) {
+            const templateName: string = workflow.spec.workflowTemplateRef.name;
+            const clusterScope: boolean = workflow.spec.workflowTemplateRef.clusterScope;
+            const url: string = clusterScope ? `/cluster-workflow-templates/${templateName}` : `/workflow-templates/${workflow.metadata.namespace}/${templateName}`;
+
+            const templateLink: Link = {
+                name: 'Workflow Template Link',
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-details/workflow-details.tsx:139]: Please change to “Open Workflow Template” and use the same icon as left-hand-nav does.

File: ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx (hunk 1 above)
Comment (shubham4443) [reply ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx:162]: Yeah, its amazing 😄

**Comment by alexec** [comment]
thank you for changes - please can I ask you attach updated screenshots? this shows it works

**Comment by shubham4443** [comment]
For WorkflowTemplate kind -

image

For ClusterWorkflowTemplate kind -

image

I've updated the first comment's screenshot as well.

**Review by alexec — APPROVED** [review]

**Pull request merged**